app/models/pending_induction_submission_batch.rb

```diff
@@ -0,0 +1,198 @@
+class PendingInductionSubmissionBatch < ApplicationRecord
+  # Class methods
+  # @return [PendingInductionSubmissionBatch] type "claim"
+  def self.new_claim_for(appropriate_body:, **)
+    new(appropriate_body:, batch_type: 'claim', **)
+  end
+
+  # @return [PendingInductionSubmissionBatch] type "action"
+  def self.new_action_for(appropriate_body:, **)
+    new(appropriate_body:, batch_type: 'action', **)
+  end
+
+  # @return [Class] A class that represents a row in the CSV file
+  # @param columns [Array<Symbol>] Array of CSV column headings
+  def self.build_row_class(columns)
+    Data.define(*columns) do
```

Comment:
This feels very magical 🪄

Maybe it would be worth moving to a concern or helper. Something like`RowBuilder` could help make the intent clearer and keep the model focussed.

Reply:
Anything I can still do to demystify it in my documentation?